### DISCUSSION TITLE
Is there any way to get the client's IP address?

### DISCUSSION THREAD

**Discussion by tksuns12** [serverpod/serverpod#362] [Q&A] [answered]
I try to detect where the client accessed to the server by its IP address. Is there any way to do that in Serverpod?

**Comment by vlidholt** [comment, marked as answer]
Yes!

You can access it through the `Session` object as such:

```dart
(session as MethodCallSession).httpRequest.remoteIpAddress
```

It should work through most load balancers and firewalls too.

**Reply by tksuns12** [reply]
Thank you very much!
